### ISSUE TITLE
Add support for a signature separator, typically '-- \n'

### ISSUE THREAD

**Issue by mrvdb** [astroidmail/astroid#371]
If I want to use a signature separator I have to include it in my signature file at the moment, which is arguably the wrong place.

If there was support for a signature separator, this would also be helpful for:
- perhaps help resolve #234 
- in thread view, anything after the signature separator could be rendered different from the main body, or left out even

**Comment by gauteh** [comment]
Marcel van der Boom writes on juli 25, 2017 12:01:
> If I want to use a signature separator I have to include it in my signature file at the moment, which is arguably the wrong place.
> 
> If there was support for a signature separator, this would also be helpful for:
> - perhaps help resolve #234 
> - in thread view, anything after the signature separator could be rendered different from the main body, or left out even
> 

Hi,

We used to have the standard signature separator. However, I think it is 
more flexible this way. People use all kinds of signatures, and 
because of this we can never rely on parsing an e-mail backwards to 
really find the signature, since this string could easily appear several 
times in the signature.

To fix #234 we should use the Xapian metadata properties which are available in notmuch now and add them to the draft. Then we can store state information there untill the draft is sent.

I agree with the final point, also this will be kind of hard to parse - 
but if we can come up with a consistent algorithm that detects what part 
is signature with a very low degree of false positives I think it should 
be included.



Regards, Gaute

**Comment by mrvdb** [comment]
What flexibility do we lose if there was, say, a config item `accounts.myaccount.signature_separator` ? 
(defaulting to "" I guess)

**Comment by gauteh** [comment]
Marcel van der Boom writes on juli 25, 2017 12:33:
> What flexibility do we lose if there was, say, a config item `accounts.myaccount.signature_separator` ? 
> (defaulting to "" I guess) 

Would it not be easier to just edit the signature file in that case?

**Comment by mrvdb** [comment]
If you're using just one MUA, yes, perhaps.

**Comment by gauteh** [comment]
Marcel van der Boom writes on juli 25, 2017 12:37:
> If you're using just one MUA, yes, perhaps.

How does mutt and others do it? If the '--  \n' is prepended by default 
I'd be more positive to adding it as config option. In that case it 
should be on by default.

**Comment by mrvdb** [comment]
my gnus and thunderbird config default to having it prepended to the signature file.  Thunderbird can be made to not do that, but havent looked at gnus yet (I'm sure it's possible)

**Comment by gauteh** [comment]
Marcel van der Boom writes on juli 25, 2017 12:41:
> my gnus and thunderbird config default to having it prepended to the signature file.  Thunderbird can be made to not do that, but havent looked at gnus yet (I'm sure it's possible)

Ok, I think the cleanest would be to have a boolean config option 
similar to what you suggest:

  accounts.myaccount.signature_separate : true

which prepends '--  \n' (note the two spaces). Anything else can be 
configured in the signature file. This will only work with inline 
signatures.

**Comment by mrvdb** [comment]
I'll create a PR for this. 

I don't think we want 2 spaces though, that will make MUAs interpret it differently. (just confirmed this in thunderbird)

Why do you want two spaces? I'm using https://www.ietf.org/rfc/rfc3676.txt section 4.3 as a reference.

**Comment by gauteh** [comment]
Marcel van der Boom writes on juli 25, 2017 13:54:
> I'll create a PR for this. 
> 
> I don't think we want 2 spaces though, that will make MUAs interpret it differently. (just confirmed this in thunderbird)
> 
> Why do you want two spaces? I'm using https://www.ietf.org/rfc/rfc3676.txt section 4.3 as a reference.

Ah right, then I misremember the spec. Thought it was two spaces.

Thanks, gaute

**Comment by mxmehl** [comment]
> in thread view, anything after the signature separator could be rendered different from the main body, or left out even

That would be great but I'm not sure we need the config file for that. `-- ` is the standard for signature files, so could astroid just use a grey font colour for all text after the last `-- ` in a message?

**Comment by gauteh** [comment]
Max writes on juli 25, 2017 14:41:
>> in thread view, anything after the signature separator could be rendered different from the main body, or left out even
> 
> That would be great but I'm not sure we need the config file for that. `-- ` is the standard for signature files, so could astroid just use a grey font colour for all text after the last `-- ` in a message?
> 

Yup, agreed - it could even be made hideable. The config option is 
only for the sending (check #372).

The '-- ' parser could work in a similar fashion as patch and code 
highlight works now. It should probably be disabled if it is a patch, 
think some of git delimiters would match.

**Comment by gauteh** [comment]
Fixed in #372 by @mrvdb.
